fix(docs): deprecate docs on localFiles, and introduce input and output files (#1584)
kestra-io/docs: content/docs/08.developer-guide/07.scripts/output-directory.md

---
title: Input and Output Files
icon: /docs/icons/dev.svg
---

## Input Files

> You can provide input files to the task using the `inputFiles` property.

Reviewer: ```suggestion
You can pass additional files to any script or CLI task using the `inputFiles` property:
```

```yaml
id: ansible
namespace: company.team

tasks:
  - id: ansible_task
    type: io.kestra.plugin.ansible.cli.AnsibleCLI
    inputFiles:
      inventory.ini: |
        localhost ansible_connection=local
      myplaybook.yml: |
        ---
        - hosts: localhost
          tasks:
            - name: Print Hello World
              debug:
                msg: "Hello, World!"
    docker:
      image: cytopia/ansible:latest-tools
    commands:
      - ansible-playbook -i inventory.ini myplaybook.yml
```

> You can also read the files from the namespace using [Namespace Files](https://kestra.io/docs/developer-guide/namespace-files).

Reviewer: ```suggestion
You can also leverage [Namespace Files](https://kestra.io/docs/developer-guide/namespace-files) as follows:
```

    type: io.kestra.plugin.ansible.cli.AnsibleCLI
    namespaceFiles:
      enabled: true
    inputFiles:
      inventory.ini: "{{ read('inventory.ini') }}"
      myplaybook.yml: "{{ read('myplaybook.yml') }}"
    docker:
      image: cytopia/ansible:latest-tools
    commands:
      - ansible-playbook -i inventory.ini myplaybook.yml
```

### Using input files to pass data from a trigger to a script task

Another use case for input files is when your custom scripts need input coming from other tasks or triggers.

Consider the following example flow that runs when a new object with the prefix `"raw/"` arrives in the S3 bucket `"declarative-orchestration"`:

```yaml
id: s3TriggerCommands
namespace: company.team
description: process CSV file from S3 trigger

tasks:
  - id: wdir
    type: io.kestra.plugin.core.flow.WorkingDirectory
    inputFiles:
      data.csv: "{{ trigger.objects | jq('.[].uri') | first }}"
    outputFiles:
      - "*.csv"
      - "*.parquet"
    tasks:
      - id: cloneRepo
        type: io.kestra.plugin.git.Clone
        url: https://github.com/kestra-io/examples
        branch: main

      - id: python
        type: io.kestra.plugin.scripts.python.Commands

>         description: this script reads a file `data.csv` from S3 trigger

Reviewer: ```suggestion
        description: this script reads a file `data.csv` from the S3 trigger
```

>         docker:
>           image: ghcr.io/kestra-io/pydata:latest

Reviewer: ```suggestion
        taskRunner:
          type: io.kestra.plugin.scripts.runner.docker.Docker
        containerImage: ghcr.io/kestra-io/pydata:latest
```

    type: io.kestra.plugin.scripts.python.Script
    taskRunner:
      type: io.kestra.plugin.core.runner.Process
    outputFiles:
      - myfile.txt
    script: |

>       f = open("myfile.txt", "a")
>       f.write("Hi, this is output from a script 👋")
>       f.close()

Reviewer: ```suggestion
      with open("myfile.txt", "a") as f:
          f.write("Hi, this is output from a script 👋")
```
